rest/ProductService.js

```diff
@@ -82,11 +88,21 @@ class ProductService {
             throw new IllegalArgumentError(INVALID_QUANTITY_ERR);
         }
     }
+    
 
-    /**
-     * TODO
-     * Validate serial numbers
-     */
+    #assertProductIsActive(productId) {
+        const product = this.#productPersistence.getById(productId);
+        if (!product.isActive) {
+            throw new Error(INVALID_IS_ACTIVE_ERR);
+        }
+    }
+
+    #assertNoAssociatedOrders(productId) {
+        const orders = this.#orderPersistence.getByProductId(productId);
+        if (orders.length > 0) {
```

Comment:
You can just check for any value returned by getByProductId here:

`#assertNoAssociatedOrders(productId) {
    if (this.#orderPersistence.getByProductId(productId).length > 0) {
        throw new Error(INVALID_ASSOCIATED_ORDER);
    }
}`

Reply:
Oops i pasted the wrong code, I mean that you can just check if the value of length is truth (other than 0)

`#assertNoAssociatedOrders(productId) {
    if (!this.#orderPersistence.getByProductId(productId).length) {
        throw new Error(INVALID_ASSOCIATED_ORDER);
    }
}`